### PR TITLE
Make email receipt upload case insensitive

### PR DESCRIPTION
**Pull request by cjdenio**
Fixes #2639

### CODE REVIEW

File: app/mailboxes/receipt_uploads_mailbox.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def process
   private
 
   def user
-    @user ||= User.find_by email: mail.from
+    @user ||= User.find_by("email ILIKE ?", User.sanitize_sql_like(mail.from[0]))
```
Comment (maxwofford) [inline app/mailboxes/receipt_uploads_mailbox.rb]: If these are already being stored in downcase only, could we just do `User.find_by email: mail.from.downcase`?

Comment (cjdenio) [reply app/mailboxes/receipt_uploads_mailbox.rb]: Hmm, that's true, though I'm a little worried about writing code that relies on emails being all-lowercase

Comment (maxwofford) [reply app/mailboxes/receipt_uploads_mailbox.rb]: User emails are stored in downcase https://github.com/hackclub/bank/blob/eb0129324841730c660578018ad97ba4231200be/app/models/organizer_position_invite.rb#L54

File: app/mailboxes/receipt_uploads_mailbox.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ def process
   private
 
   def user
-    @user ||= User.find_by email: mail.from
+    @user ||= User.find_by(email: mail.from[0].downcase)
```
Comment (maxwofford) [inline app/mailboxes/receipt_uploads_mailbox.rb]: How was this code running with `mail.from` before if `mail.from[0]` is required now?

Comment (cjdenio) [reply app/mailboxes/receipt_uploads_mailbox.rb]: `mail.from` is an array, and `find_by` accepts an array (e.g. `User.find_by(email: ["email1@gmail.com", "email2@gmail.com"])`)

Comment (cjdenio) [reply app/mailboxes/receipt_uploads_mailbox.rb]:
```ruby
mail.from.map(&:downcase)
```

could also work, but I think the `From` header can only contain one value

**Review by maxwofford — APPROVED** [review]
LGTM!

**Pull request merged**